quinn-proto/src/connection/mod.rs

```diff
@@ -477,13 +482,17 @@ impl Connection {
                     SpaceId::Data,
                     "PATH_CHALLENGE queued without 1-RTT keys"
                 );
-                let mut buf = BytesMut::with_capacity(self.path.current_mtu() as usize);
-                let buf_capacity = self.path.current_mtu() as usize;
+                let remaining_capacity = buf.capacity() - buf.len();
+                if remaining_capacity < self.path.current_mtu() as usize {
+                    buf.reserve(self.path.current_mtu() as usize - remaining_capacity);
+                }
```

Comment:
`BytesMut::reserve` is relative to the current length, not the current capacity (see [source](https://github.com/tokio-rs/bytes/blob/bde8c50703869f54b905560eb62c2478a0111885/src/bytes_mut.rs#L566-L574)), so this can under-reserve. Further, we're actually duplicating the short-circuit logic inside `reserve` here; we might as well rely on that directly for simplicity.

```suggestion
                buf.reserve(self.path.current_mtu() as usize);
```